envoy/common/key_value_store.h

```diff
@@ -0,0 +1,41 @@
+#pragma once
+
+#include "envoy/common/pure.h"
+
+#include "absl/strings/string_view.h"
+
+namespace Envoy {
+
+// A key value store, designed to periodically flush key value pairs to long
+// term storage (disk or otherwise)
+class KeyValueStore {
+public:
+  virtual ~KeyValueStore() {}
+
+  /**
+   * Adds or updates a key:value pair in the store.
+   * @param key supplies a key to add or update.
+   * @param value supplies the value to set for that key.
+   */
+  virtual void addOrUpdateKey(absl::string_view key, absl::string_view value) PURE;
+
+  /**
+   * Removes a key:value pair from the store. This is a no-op if the key is not present.
+   * @param key supplies a key to remove from the store.
+   */
+  virtual void removeKey(absl::string_view key) PURE;
+
+  /**
+   * Returns the value of the key provided.
+   * @param key supplies a key to return the value of.
+   * @return the value, if the key is in the store, empty string otherwise.
+   */
+  virtual absl::string_view getKey(absl::string_view key) PURE;
```

Comment:
nit: This method returns the value (as the comment said) but the name suggests it's returning a key (getKey). I think that for all 3 of these methods, I'd remove Key from the name:

addOrUpdateKey(key, value) => addOrUpdate(key, value)
removeKey(key) => remove(key)
value = getKey(key) => value = get(key)

Reply:
done